src/core/EdFi.DataManagementService.Core/ApiService.cs

```diff
@@ -264,4 +266,107 @@ public IList<IDataModelInfo> GetDataModelInfo()
         }
         return result;
     }
+
+    public JsonArray GetDependencies()
+    {
+        JsonNode schema = _apiSchemaProvider.ApiSchemaRootNode;
+
+        KeyValuePair<string, JsonNode?>[]? projectSchemas = schema["projectSchemas"]?.AsObject().ToArray();
+        if (projectSchemas == null || projectSchemas.Length == 0)
+        {
+            string errorMessage = "No projectSchemas found, ApiSchema.json is invalid";
+            _logger.LogCritical(errorMessage);
+            throw new InvalidOperationException(errorMessage);
+        }
+
+        List<JsonNode> projectSchemaNodes = projectSchemas
+            .Where(x => x.Value != null)
+            .Select(x => x.Value ?? new JsonObject())
+            .ToList();
+
+        var dependenciesJsonArray = new JsonArray();
+        foreach (JsonNode projectSchemaNode in projectSchemaNodes)
+        {
+            KeyValuePair<string, JsonNode?>[]? resourceSchemas =
+                projectSchemaNode?["resourceSchemas"]?.AsObject().ToArray();
+            Debug.Assert(resourceSchemas != null, nameof(resourceSchemas) + " != null");
+
+
+            Dictionary<string, List<string>> dependencies =
+                resourceSchemas.ToDictionary(rs => rs.Value!["resourceName"]!.GetValue<string>(), rs => new List<string>());
+
+            foreach (var resourceSchema in resourceSchemas!.Select(rs => rs.Value))
+            {
+                var resourceSchemaJsonObject = resourceSchema;
+                var documentPathsMappingObj = resourceSchemaJsonObject?["documentPathsMapping"]?.AsObject();
+                if (documentPathsMappingObj != null)
+                {
+                    var references = documentPathsMappingObj.Select(dmo => dmo.Value?.AsObject()).Where(o =>
+                    {
+                        Debug.Assert(o != null, nameof(o) + " != null");
+                        return (bool)(o["isReference"] ?? false);
+                    });
+
+                    foreach (var reference in references)
+                    {
+                        dependencies[resourceSchemaJsonObject!["resourceName"]!.GetValue<string>()].Add(reference!["resourceName"]!.GetValue<string>());
+                    }
+                }
+            }
+            List<KeyValuePair<string, int>> orderedResources = new List<KeyValuePair<string, int>>();
+            Dictionary<string, int> visitedResources = new Dictionary<string, int>();
+            foreach (var dependency in dependencies.OrderBy(d => d.Value.Count).ThenBy(d => d.Key).Select(d => d.Key))
+            {
+                RecursivelyDetermineDependencies(dependency, 0);
+            }
+
+            int RecursivelyDetermineDependencies(string resourceName, int depth)
+            {
+
+                // some hacks
+                if (resourceName == "EducationOrganization")
+                {
+                    resourceName = "School";
+                }
```

Comment:
Really?

Reply:
Open to suggestions here. Definitely a code smell.

Reply:
It's a workaround for a small lack of information in the ApiSchema.json. One option is to have MetaEd include this information in the existing `abstractResources` section of ApiSchema.json

Reply:
Here's an idea. Read the `abtractResources` section to get `EducationOrganization` and `GeneralStudentProgramAssociation`. Then scan the resources for the first one that has `isSubclass: true` and matches each. Use that instead.

It might work, or it might be too simplistic in that it picks e.g. `EducationOrganizationNetwork` which isn't as "connected" (from a dependency graph perspective) as `School`. In which case you'd need to do some sort of "connectedness" analysis of subclasses to select the one to substitute.

Reply:
Another idea, ignore the abstract superclasses to start and get an ordering. Then find all the subclasses in the ordering and insert the superclass right after the last subclass. That's probably the easiest way.